pkg/controllers/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -535,6 +401,121 @@ func (d *DependenciesDistributor) removeScheduleResultFromAttachedBindings(bindi
 	return utilerrors.NewAggregate(errs)
 }
 
+func (d *DependenciesDistributor) createOrUpdateAttachedBinding(attachedBinding *workv1alpha2.ResourceBinding) error {
+	if err := d.Client.Create(context.TODO(), attachedBinding); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			klog.Infof("Failed to create resource binding(%s/%s): %v", attachedBinding.Namespace, attachedBinding.Name, err)
+			return err
+		}
+
+		existBinding := &workv1alpha2.ResourceBinding{}
+		key := client.ObjectKeyFromObject(attachedBinding)
+		if err := d.Client.Get(context.TODO(), key, existBinding); err != nil {
+			klog.Infof("Failed to get resource binding(%s/%s): %v", attachedBinding.Namespace, attachedBinding.Name, err)
+			return err
+		}
+
+		updatedBindingSnapshot := mergeBindingSnapshot(existBinding.Spec.RequiredBy, attachedBinding.Spec.RequiredBy)
+		existBinding.Spec.RequiredBy = updatedBindingSnapshot
+		existBinding.Labels = util.DedupeAndMergeLabels(existBinding.Labels, attachedBinding.Labels)
+		existBinding.Spec.Resource = attachedBinding.Spec.Resource
+
+		if err := d.Client.Update(context.TODO(), existBinding); err != nil {
+			klog.Errorf("Failed to update resource binding(%s/%s): %v", existBinding.Namespace, existBinding.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+// Start runs the distributor, never stop until stopCh closed.
+func (d *DependenciesDistributor) Start(ctx context.Context) error {
+	klog.Infof("Starting dependencies distributor.")
+	d.stopCh = ctx.Done()
+
+	bindingWorkerOptions := util.Options{
+		Name:          "resourceBinding reconciler",
+		KeyFunc:       detector.ClusterWideKeyFunc,
+		ReconcileFunc: d.ReconcileResourceBinding,
+	}
+	// setup binding reconcile worker
+	d.bindingReconcileWorker = util.NewAsyncWorker(bindingWorkerOptions)
+	d.bindingReconcileWorker.Run(2, d.stopCh)
```

Comment:
> The main job of the controller is responsible to reconcile RB changes and sync schedule results to dependencies( RBs referencing dependencies like ConfigMap/Secret). That means we don't need the bindingReconcileWorker anymore.


Yes, I also considered removing `bindingReconcileWorker`, but when I saw L443 `d.bindingReconcileWorker.Run(2, d.stopCh)`, I chose not to remove it because I understand that `bindingReconcileWorker` can be more efficient than the `formal controller`.